src/fieldValidation.ts

```diff
@@ -0,0 +1,30 @@
+import { BaseFieldDataType } from './types';
+
+export const fieldValueIsValid = (
+	fieldValue: string,
+	dataType: BaseFieldDataType,
+) => {
+	switch (dataType) {
+		case BaseFieldDataType.NUMBER:
+			return /^[0-9]*$/.test(fieldValue);
+			break;
+		case BaseFieldDataType.PHONE_NUMBER:
+			return /^\s*(?:\+?(\d{1,3}))?[-. (]*(\d{3})[-. )]*(\d{3})[-. ]*(\d{4})(?: *x(\d+))?\s*$/.test(
+				fieldValue,
+			);
+			break;
+		case BaseFieldDataType.EMAIL:
+			return /^\S+@\S+$/.test(fieldValue);
+			break;
+		case BaseFieldDataType.URL:
+			return /^(https?:\/\/)?(www\.)?[-a-zA-Z0-9@:%._+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_+.~#?&//=]*)/.test(
```

Comment:
Just a note that while I think it's totally OK to merge this as is -- I probably would have preferred this PR only add the most basic validation (number and string), and in a future PR add more complex ones using libraries rather than introducing code that is complex enough to be trouble but not tested enough to be necessarily right!

Using a library here allows us to avoid having to write sweeping unit tests that test all the weird edge cases -- we can rely on the project to have correct validation!

TL;DR I think we need to replace this regex with real validation tools / libraries as soon as possible.

Phone numbers could be: https://www.npmjs.com/package/phone, (or maybe there is an ajv utility for it)

URLs and Emails would be best done using `ajv-formats` https://github.com/ajv-validator/ajv-formats

---

We should make an issue for these (and prioritize addressing next week IMHO!)

Reply:
@hminsky2002 just to clarify (since this is marked as resolved) -- can you make those issues (then you can mention them as a comment here and github will link to the context)

Reply:
Issues #875 and #874